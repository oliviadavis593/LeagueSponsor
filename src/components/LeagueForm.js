import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import Logo from '../imgs/leaguelogo.png';


class LeagueForm extends Component {
    render() {
        return(
            <div>
                <div>
                    <NavLink to='/dashboard'>
                        <img src={Logo} alt='ls-logo' className='ls-logo' />
                    </NavLink>
                </div>
                <main>
                    <form>

                    </form>
                </main>
            </div>
        )
    }
}

export default LeagueForm; 