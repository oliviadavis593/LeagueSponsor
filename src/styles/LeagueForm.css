.ls-leagueform {
    width: 300px;
    height: 600px; 
    margin: 0 auto; 
    border: 1px solid #233142;
    background-color: #233142; 
    color: #fff; 
}

fieldset {
    border: none; 
}

legend {
    font-size: 20px; 
}

legend p {
    font-size: 20px; 
}

label {
    font-size: 22px; 
}

input {
    display: flex; 
    margin: 10px auto; 
    width: 200px; 
    height: 20px; 
}

.ls-form__button {
    width: 100px; 
    height: 40px; 
    font-family: 'Oswald', sans-serif;
    background-color: #def3fa;
    border: none; 
    font-size: 20px; 
}

