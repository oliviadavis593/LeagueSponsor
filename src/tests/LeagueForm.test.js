import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import LeagueForm from '../components/LeagueForm';

describe('LeagueForm Component', () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><LeagueForm /></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer 
            .create(<BrowserRouter><LeagueForm /></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})