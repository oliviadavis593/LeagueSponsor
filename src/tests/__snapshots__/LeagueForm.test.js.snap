// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeagueForm Component renders the UI as expected 1`] = `
<div>
  <div>
    <a
      aria-current={null}
      href="/dashboard"
      onClick={[Function]}
    >
      <img
        alt="ls-logo"
        className="ls-logo"
        src="leaguelogo.png"
      />
    </a>
  </div>
  <main>
    <form
      className="ls-leagueform"
      onSubmit={[Function]}
    >
      <fieldset>
        <legend>
          <h2>
            League Form
          </h2>
          <p>
            Add a league to our platform so sponsorships can see you!
          </p>
        </legend>
        <label
          htmlFor="league_name"
        >
          League Name:
        </label>
        <input
          id="league_name"
          name="league_name"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          htmlFor="latitude"
        >
          Latitude:
        </label>
        <input
          id="latitude"
          name="location"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          htmlFor="longitude"
        >
          Longitude:
        </label>
        <input
          id="longitude"
          name="longitude"
          onChange={[Function]}
          required={true}
          type="number"
        />
        <label
          htmlFor="website"
        >
          Website:
        </label>
        <input
          id="website"
          name="website"
          onChange={[Function]}
          type="text"
        />
        <label
          htmlFor="price"
        >
          Price:
        </label>
        <input
          id="price"
          name="price"
          onChange={[Function]}
          required={true}
          type="number"
        />
      </fieldset>
      <button
        className="ls-form__button"
        type="submit"
      >
        Add League
      </button>
    </form>
  </main>
</div>
`;
